tor_ocr/main.py

```diff
@@ -130,20 +135,33 @@ def decode_image_from_url(url, overlay=False, api_key=__OCR_API_KEY__):
         'apikey': api_key,
     }
 
-    result = requests.post(
-        __OCR_API_URL__,
-        data=payload,
-    )
-
-    if not result.ok:
-        logging.error(
-            f'ERROR {result.status_code} with OCR:\n\nHEADERS:\n '
-            f'{repr(result.headers)}\n\nBODY:\n{repr(result.text)} '
+    result = None
+
+    for API in __OCR_API_URLS__:
+        try:
+            result = requests.post(API, data=payload)
+            # crash and burn if the API is down, or similar :)
+            result.raise_for_status()
+
+            # if the request succeeds, we'll have a result. Therefore, just
+            # break the loop here.
+            break
+        except ConnectionError:
+            # try the next API in the list, then release from the loop if we
+            # exhaust our options.
+            pass
+        except RequestException:
+            # we have a result object here but it's not right.
+            logging.error(
+                f'ERROR {result.status_code} with OCR:\n\nHEADERS:\n '
+                f'{repr(result.headers)}\n\nBODY:\n{repr(result.text)} '
+            )
+
+    if result is None or not result.ok:
+        raise ConnectionError(
+            'Attempted all three OCR.space APIs -- cannot connect!'
```

Comment:
Nit: replace `APIs -- cannot connect!` with `APIs—cannot connect!` (em-dash)

Reply:
I'll think about it.

(hint: probably not making this change 😛 )

Reply:
😂 fair enough—I just really like em-dash 😉